src/databricks/labs/ucx/azure/resources.py

```diff
@@ -222,23 +223,26 @@ def __init__(self, host_endpoint: str, service_endpoint: str):
         self.api_client = ApiClient(
             Config(
                 host=host_endpoint,
-                credentials_provider=self._provider_for(service_endpoint),
+                credentials_strategy=self._provider_for(service_endpoint),
             )
         )
         self._token_source = AzureCliTokenSource(host_endpoint)
 
-    def _provider_for(self, endpoint: str):
-        @credentials_provider("azure-cli", ["host"])
-        def _credentials(_: Config):
+    @staticmethod
+    def _provider_for(endpoint: str) -> CredentialsStrategy:
+        @credentials_strategy("azure-cli", ["host"])
+        def _credentials(_: Config) -> CredentialsStrategy:
             token_source = AzureCliTokenSource(endpoint)
 
             def inner() -> dict[str, str]:
                 token = token_source.token()
                 return {"Authorization": f"{token.token_type} {token.access_token}"}
 
-            return inner
+            # The type hints are off but according to the sdk:
+            # https://github.com/databricks/databricks-sdk-py/blob/f7d920e1f912b204669b826ed76e026607c59797/databricks/sdk/credentials_provider.py#L116
+            return inner  # type: ignore
```

Comment:
Mentioned this here https://github.com/databricks/databricks-sdk-py/pull/644#issuecomment-2196533715